src/ActiveRecord.php

```diff
@@ -153,71 +186,121 @@ public static function primaryKey() {
+     * ```
+     *
+     * @param array        $attributes Attribute values (name-value pairs) to be saved into the table.
+     * @param string|array $condition  The conditions of the rows.
+     * @return void
+     * @throws \yii\base\NotSupportedException Not implemented yet.
      */
-    private static function createQueryWithParameter($options = null) {
-        $query = self::find();
-        if ($options !== null) {
-            foreach($options as $attribute => $value) {
-                $query->{$attribute} = $value;
-            }
-        }
-        return $query;
+    public static function updateAll(array $attributes, $condition = '')
+    {
+        throw new \yii\base\NotSupportedException(__METHOD__ . ' is not supported.');
     }
 
     /**
-     * @inheritdoc
-     * @todo
+     * Updates the whole table using the provided counter changes and conditions.
+     * For example, to increment all customers' age by 1,
+     *
+     * ```php
+     * Customer::updateAllCounters(['age' => 1]);
+     * ```
+     *
+     * @param array        $counters  The counters to be updated (attribute name => increment value).
+     * Use negative values if you want to decrement the counters.
+     * @param string|array $condition The conditions to select the rows to be updated.
+     * @return void
+     * @throws \yii\base\NotSupportedException Not implemented yet.
      */
-    public static function updateAll($attributes, $condition = '') {
-        parent::updateAll($attributes, $condition);
+    public static function updateAllCounters(array $counters, $condition = '')
+    {
+        throw new \yii\base\NotSupportedException(__METHOD__ . ' is not supported.');
+    }
+
+    /**
+     * Deletes rows in the table using the provided conditions.
+     * WARNING: If you do not specify any condition, this method will delete ALL rows in the table.
+     *
+     * For example, to delete all customers whose status is 3:
+     *
+     * ```php
+     * Customer::deleteAll('status = 3');
+     * ```
+     *
+     * @param string|array $condition The conditions that will select the rows.
+     * @param array        $params    The parameters (name => value) to be bound to the query.
+     * @return void
+     * @throws \yii\base\NotSupportedException Not implemented yet.
+     */
+    public static function deleteAll($condition = '', array $params = [])
+    {
+        throw new \yii\base\NotSupportedException(__METHOD__ . ' is not supported.');
     }
     
     /**
-     * @inheritdoc
-     * @todo
+     * Populate active records from query response.
+     * @param ActiveQuery $query The query that produces the records.
+     * @param array $response    The operation response.
+     * @return static[]
      */
-    public static function updateAllCounters($counters, $condition = '') {
-        parent::updateAllCounters($counters, $condition);
+    public static function populateRecords(ActiveQuery $query, array $response)
```

Comment:
Still working on that. I'm literally currently thinking to remove it to the Query level instead of active record level.